src/Protocol/ActivityPub/Processor.php

```diff
@@ -94,39 +94,43 @@ private static function constructAttachList($activity, $item)
 		}
 
 		foreach ($activity['attachments'] as $attach) {
-			$filetype = strtolower(substr($attach['mediaType'], 0, strpos($attach['mediaType'], '/')));
-			if ($filetype == 'image') {
-				if (!empty($activity['source']) && strpos($activity['source'], $attach['url'])) {
-					continue;
-				}
-
-				if (empty($attach['name'])) {
-					$item['body'] .= "\n[img]" . $attach['url'] . '[/img]';
-				} else {
-					$item['body'] .= "\n[img=" . $attach['url'] . ']' . $attach['name'] . '[/img]';
-				}
-			} elseif ($filetype == 'audio') {
-				if (!empty($activity['source']) && strpos($activity['source'], $attach['url'])) {
-					continue;
-				}
-
-				$item['body'] .= "\n[audio]" . $attach['url'] . '[/audio]';
-			} elseif ($filetype == 'video') {
-				if (!empty($activity['source']) && strpos($activity['source'], $attach['url'])) {
-					continue;
-				}
-
-				$item['body'] .= "\n[video]" . $attach['url'] . '[/video]';
-			} else {
-				if (!empty($item["attach"])) {
-					$item["attach"] .= ',';
-				} else {
-					$item["attach"] = '';
-				}
-				if (!isset($attach['length'])) {
-					$attach['length'] = "0";
-				}
-				$item["attach"] .= '[attach]href="'.$attach['url'].'" length="'.$attach['length'].'" type="'.$attach['mediaType'].'" title="'.($attach['name'] ?? '') .'"[/attach]';
+			switch ($attach['type']) {
+				case 'link':
+					$item['body'] .= "\n[attachment type='link' url='" . $attach['url'] . "' title='" . ($attach['title'] ?? '') . "' image='" . ($attach['image'] ?? '') . "']" . ($attach['desc'] ?? '') . '[/attachment]';
```

Comment:
The `[attachment ...]` BBCode can only be used exactly one time in a BBCode document. Means we have to ensure that there is no other `[attachment ...]` element in the code when adding this.

Also in the future we might to send this data as well by ourselves. Then we (most likely) would already had this `[attachment ...]` element in the `$activity['source']` variable. I guess this is a similar situation like with the other types here in this function.

Reply:
Let's say there are multiple Page attachment, what should we do with the rest after one has been used as `[attachment]`?

Reply:
This is a good question. We possibly could transform it into a regular link. But we have to ensure that the `[attachment]` element is the last element of the post. It doesn't look nice if there is some more content after this.

Reply:
What would generate an `[attachment]` in incoming AP posts except this code? A Friendica post with an `[attachment]` in the body already? But it wouldn't have a `type:Page` attachment either.

So I feel like I can loop through them and transform the last one as an attachment and add qualified links for the others.

Reply:
We currently don't generate this type of attachment - but could in the future. So we have to check for existing `[attachment]` elements in `$activity['source']` as well.